### PR TITLE
Move asset pipeline javascripts to frontend

### PR DESCRIPTION
**Pull request by oliverguenther**
### ToDo

- [x] Move global JS files to frontend/src/...
  - [x] Get frontend assets green (npm run serve)
- [x] Remove/Replace .js.erb files
- [x] Remove obsolete .js files
- [x] Load all plugin .js files correctly
  - [x] Plugins which are already under frontend/modules
  - [x] Subscription plugin (done in https://github.com/finnlabs/saas-openproject/pull/74)
- [x] Finish all Todos (marked like that: Todo for task #33507)

### CODE REVIEW

**Comment by HDinger** [comment]
Ticket:
https://community.openproject.com/projects/openproject/work_packages/33507/activity

File: frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 
 export const reportingPageComponentSelector = 'op-reporting-page';
 
+import './functionality/reporting_engine';
+import './functionality/reporting_engine/filters';
+import './functionality/reporting_engine/group_bys';
+import './functionality/reporting_engine/restore_query';
+import './functionality/reporting_engine/controls';
+
+
```
Comment (openprojectci) [inline frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts:12]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/7c43ca799794047bcf272d9ecda248b84d763c76/frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

Comment (openprojectci) [inline frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts:12]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/743dd4d4e3903df104c4b7f27c4d62f6a5c6bdc8/frontend/src/app/modules/reporting/reporting-page/reporting-page.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/743dd4d4e3903df104c4b7f27c4d62f6a5c6bdc8/frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L33">frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L33</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**